### PR TITLE
feat: add C implementation for `math/base/assert/uint32-is-pow2`

### LINKED ISSUE #1868: [RFC]: add C implementation for `@stdlib/math/base/assert/uint32-is-pow2`
### Description

This RFC proposes adding native C implementation for math/base/assert/uint32-is-pow2: 
```javascript
bool stdlib_base_uint32_is_pow2( const uint32_t x );
```

### Related Issues

N/A

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by performant23**
This commit if applied adds the C implementation to `@stdlib/math/base/assert/uint32-is-pow2` package

## Description

This pull request:

- adds the native C implementation to `@stdlib/math/base/assert/uint32-is-pow2`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1868 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by performant23** [comment]
Regarding the naming convention: Since `benchmark.js` uses the alias, `isPow2`, it is used in benchmark.native.js and since  in `test.js` it is used as `isPow2Uint32`, the same alias is used for test.native.js

File: lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/README.md [hunk 1]
```diff
@@ -72,6 +72,98 @@ for ( i = 0; i < 100; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/assert/uint32_is_pow2.h"
+```
+
+#### stdlib_base_uint32_is_pow2( x )
+
+Test whether an unsigned integer is a power of 2
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/README.md]:
```suggestion
Tests whether an unsigned integer is a power of 2
```

File: lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/README.md [hunk 2]
```diff
@@ -72,6 +72,98 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdint.h>
+
+int main( void ) {
+    const uint32_t x[] = { 0, 1, 2, 3, 4, 5, 8, 10, 16, 1024 };
+    bool b;
+    int i;
+
+    for ( i = 0; i < 9; i++ ) {
+        b = stdlib_base_uint32_is_pow2( x[ i ] );
+        printf( "Value: %d. is a power of 2? %s.\n", x[ i ], ( b ) ? "True" : "False" );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/README.md]:
```suggestion
        printf( "Value: %u. is a power of 2? %s.\n", x[ i ], ( b ) ? "True" : "False" );
```

File: lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#include <stdint.h>
+
+int main( void ) {
+	const uint32_t x[] = { 0, 1, 2, 3, 4, 5, 8, 10, 16, 1024 };
+	bool b;
+	int i;
+
+	for ( i = 0; i < 9; i++ ) {
+		b = stdlib_base_uint32_is_pow2( x[ i ] );
+		printf( "Value: %d. is a power of 2? %s.\n", x[ i ], ( b ) ? "True" : "False" );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/examples/c/example.c]:
```suggestion
		printf( "Value: %u. is a power of 2? %s.\n", x[ i ], ( b ) ? "True" : "False" );
```

File: lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/lib/native.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+// MODULES //
+
+var Boolean = require( '@stdlib/boolean/ctor' );
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Test whether an unsigned integer is a power of 2.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/lib/native.js]:
```suggestion
* Tests whether an unsigned integer is a power of 2.
```

Comment (performant23) [reply lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/lib/native.js]: Hi Pranav, thank you so much for your feedback! Should I add the required changes or are you working on reviewing the remaining files?

File: lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/src/main.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+* @return     output value
+*
+* @example
+* #include <stdbool.h>
+*
+* bool out = stdlib_base_uint32_is_pow2( 3 );
+* // returns false
+*/
+bool stdlib_base_uint32_is_pow2( const uint32_t x ) {
+	return (x && !(x & (x - 1)));
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/src/main.c]:
```suggestion
	uint32_t xc;
	xc = x >> 0;
	return ( !!xc && !( xc & ( xc - 1 ) ) );
```

**Review by Pranavchiku — APPROVED** [review]
Works well locally, thanks @performant23 !

File: lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/README.md [hunk 6]
```diff
@@ -72,6 +72,98 @@ for ( i = 0; i < 100; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/assert/uint32_is_pow2.h"
+```
+
+#### stdlib_base_uint32_is_pow2( x )
+
+Tests whether an unsigned integer is a power of 2
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/README.md]:
```suggestion
Tests whether an unsigned integer is a power of 2.
```

File: lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/src/main.c [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/uint32_is_pow2.h"
+#include <stdint.h>
+
+/**
+* Test whether an unsigned integer is a power of 2.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/assert/uint32-is-pow2/src/main.c]:
```suggestion
* Tests whether an unsigned integer is a power of 2.
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**